src/sidebar/components/user-menu.js

```diff
@@ -11,15 +11,32 @@ import MenuItem from './menu-item';
 import MenuSection from './menu-section';
 import SvgIcon from '../../shared/components/svg-icon';
 
+/** @typedef {import('../../types/config').SidebarConfig} SidebarConfig */
+
+/** @typedef {import('../../types/config').HostConfig} HostConfig */
+
+/** @typedef {import('../../types/auth').AuthState} AuthState */
```

Comment:
```suggestion
/**
 * @typedef {import('../../types/config').SidebarConfig} SidebarConfig
 * @typedef {import('../../types/config').HostConfig} HostConfig
 * @typedef {import('../../types/auth').AuthState} AuthState 
 */
```

Elsewhere I've taken the approach of combining type imports into a single `/** .. */` comment block to reduce visual noise and the number of empty lines.

Reply:
I like this, thanks